src/javascript/binary_japan/symbols.js

```diff
@@ -0,0 +1,20 @@
+if(typeof is_japan === 'function'){
+	Symbols._details = Symbols.details.bind({});
+
+	Object.defineProperties(Symbols,{
+		details:{
+			value:function(data){
+				var active_symbols = [];
+
+				data.active_symbols.forEach(function(symbol){
+					if(symbol.market==='forex' && symbol.submarket==='major_pairs'){
+						active_symbols.push(symbol);
+					}
```

Comment:
And we would have combined map() + filter() to achieve this in a more functional way.